src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -618,15 +618,12 @@ static int CountDigits(int value)
         }
 
         // Some fields that have a reserved spot in the header, may not fit in such field anymore, but they can fit in the
-        // extended attributes. They get collected and saved in that dictionary, with no restrictions.
-        private void CollectExtendedAttributesFromStandardFieldsIfNeeded()
+        // extended attributes. They are always collected or updated in that dictionary, with no restrictions.
+        private void CollectExtendedAttributesFromStandardFields()
         {
-            ExtendedAttributes.Add(PaxEaName, _name);
+            ExtendedAttributes[PaxEaName] = _name;
```

Comment:
What does this mean for someone building a PaxTarEntry with their user-defined extended attributes? Are we just overwriting it? is that OK? I thought you were going to perform the set only for entries that originated from a TarReader.

Reply:
The user should change the `name`, `linkname`, `mtime`, `gname` and `uname` using the public properties we expose. 

Maybe we should have an exception that is thrown when a `PaxTarEntry` constructor that takes a dictionary has one of those 4 elements already inserted in the dictionary.

So yes, it's ok to overwrite them. If the user changes tha value of one of those 4 using the public properties, we should update the dictionary automatically, not them.

Reply:
I don't think is ok to silently overwrite them, in that case I do prefer the current `ArgumentException` to signal that those keys are reserved, or as you say, we should check up-front if the extended attributes passed to the ctor contain one of the "reserved attributes", plus, we should assert that overwrite of "reserved attributes" only occurs for `PaxTarEntry`s coming from `TarReader`.

Reply:
As discussed on our videocall, we will store a boolean that determines if the user has modified one of the reserved fields manually via a property, in which case, we need to sync its value with the dictionary right before the entry is written to an archive.